app/models/concerns/account_interactions.rb

```diff
@@ -264,6 +264,7 @@ def followers_for_local_distribution
 
   def lists_for_local_distribution
     lists.joins(account: :user)
+         .where('list_accounts.follow_id IS NOT NULL')
```

Comment:
This might require an extra index: indeed, `list_accounts` are indexed by `account_id`, and by `follow_id` separately, so the query planner would have to use two indexes. It may be worth adding an index on `account_id, follow_id` (`WHERE follow_id IS NOT NULL`).